app/src/Components/DisplayCards/DisplayCards.tsx

```diff
@@ -1,30 +1,45 @@
-import { useState } from "react";
+import { useState, useEffect } from "react";
 import ProductCard from "../ProductCard/ProductCard";
-import cardInfo from "../ProductCard/cardInfo";
 import './DisplayCards.css';
 import ReactPaginate from "react-paginate";
+import { PokemonCard } from "../../../../api/src/interfaces/cards.interface";
 
 function DisplayCards () {
   const [pageNumber, setPageNumber] = useState(0);
+  const [cards, setCards] = useState<PokemonCard[]>([]);
+  const [totalPages, setTotalPages] = useState(0);
   
-  const cardsPerPage: number = 12;
-  const pagesVisited: number = pageNumber * cardsPerPage;
-
-  const displaySetOfCards = cardInfo
-    .slice(pagesVisited, pagesVisited + cardsPerPage)
-    .map((card) => {
-      return (
-        <ProductCard 
-          name={card.name}
-          image={card.image}
-          salePrice={card.salePrice}
-          marketPrice={card.marketPrice}
-          description={card.description}
-        />
-      );
-    });
+  // 12 Cards per page
+  const limit: number = 12;
+  
+  // Update the cards each time the user changes the page
+  useEffect(() => {
+    const fetchCurrentCards = async() => {
+      try {
+        const result = await fetch(`http://localhost:3000/cards/?page=${pageNumber + 1}&limit=${limit}`);
+        const data = await result.json();
+        setCards(data.data);
+        setTotalPages(data.totalPages);
+      } catch (error) {
+        console.error(error);
+      }
+    };
+    fetchCurrentCards();
+  }, [pageNumber]);
 
-  const pageCount: number = Math.ceil(cardInfo.length / cardsPerPage);
+  // Map all of the current cards
+  const displaySetOfCards = cards
+  .map((card) => {
```

Comment:
I would change the above code to `cards && cards.map((card) => {`

Because, if we called map on the cards array when it was empty, it would just bring up an error (it happened to me when I ran the code)

https://stackoverflow.com/questions/63851344/nodejs-typeerror-err-invalid-arg-type-the-path-argument-must-be-of-type-s